### PR TITLE
xarray_utils: mask ocean and Antarctica

### PR DESCRIPTION
**Pull request by mathause**
- [ ] Closes #xxx
 - [x] Tests added
 - [x] Passes `isort . && black . && flake8`
 - [x] Fully documented, including `CHANGELOG.rst`

The second one in the series of data manipulation PRs (after #217). This adds functions to mask out the ocean and Antarctica. I think this is quite a bit easier than #217.

The naming of `x_dim` vs. `x_coords` is intentional (compare in #217). For most cases this will not play a role and the two will be the same. However, for regional simulations with a rotated grid it will make a difference. I should probably do a terminology section as for [xarray](https://docs.xarray.dev/en/stable/user-guide/terminology.html).

### CODE REVIEW

**Comment by mathause** [comment]
@jschwaab - if you have time I'd appreciate if you can take a look.

File: docs/source/api.rst [hunk 1]
```diff
@@ -36,6 +36,16 @@ Computation
...
 
+Data manipulation
+-----------------
+
+.. autosummary::
+   :toctree: generated/
+
+   ~xarray_utils.mask_ocean_fraction
+   ~xarray_utils.mask_ocean
+   ~xarray_utils.mask_antarctica
```
Comment (mathause) [inline docs/source/api.rst:47]:
```suggestion
   ~xarray_utils.mask.mask_ocean_fraction
   ~xarray_utils.mask.mask_ocean
   ~xarray_utils.mask.mask_antarctica
```

**Comment by jschwaab** [comment]
Looks great Mathias! The functions for masking out ocean and Antarctica are very clear now. My only suggestion would be to maybe include some sort of comment to make clear that the masking in `mask_land` (for irregular grids) is based on whether the center of the grid cell is over land or ocean.

I like the idea of keeping things simple using http://www.naturalearthdata.com `regionmask.defined_regions.natural_earth_v5_0_0.land_110` and not allowing for other masks

**Comment by mathause** [comment]
@jschwaab Thanks a lot for your review. This is ready for a final look.

**Review by jschwaab — APPROVED** [review]
perfect!

**Pull request merged**